### ISSUE TITLE
linter: switch-exhaustiveness-check docs describe defaultCaseCommentPattern as annotating a live default clause

### ISSUE THREAD

**Issue by shbernal** [oxc-project/oxc#26087]
### What version of Oxlint are you using?

1.79.0 (with `oxlint-tsgolint` 7.0.2001)

### What command did you run?

`oxlint . -c .oxlintrc.json`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": ["typescript"],
  "categories": {},
  "rules": {
    "typescript/switch-exhaustiveness-check": ["error", { "defaultCaseCommentPattern": "^skip default$" }]
  },
  "options": { "typeAware": true, "typeCheck": true }
}
```

### What happened?

**Disclosure: this issue was drafted with AI assistance and reviewed by me before submission, per the [AI Usage Policy](https://github.com/oxc-project/oxc/blob/main/CONTRIBUTING.md#ai-usage-policy).**

This is a documentation bug, not a behaviour bug. The rule itself is correct; the option's
documented example describes a mechanism that neither Oxlint nor typescript-eslint
implements, and following it produces a finding with no obvious explanation.

The option is documented as:

> Regular expression pattern that when matched in a default case comment, will suppress
> the exhaustiveness check.
> Example: `"@skip-exhaustive-check"` to allow `default: // @skip-exhaustive-check`

— [`crates/oxc_linter/src/rules/typescript/switch_exhaustiveness_check.rs#L20-L22`](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/typescript/switch_exhaustiveness_check.rs#L20-L22),
rendered at [oxc.rs](https://oxc.rs/docs/guide/usage/linter/rules/typescript/switch-exhaustiveness-check).

That reads as "keep your `default` arm and mark it". It does not work:

```ts
declare const value: 'a' | 'b';

switch (value) {
  case 'a':
    break;
  default: // skip default
    break;
}
```

```text
error typescript(switch-exhaustiveness-check): Switch is not exhaustive. Cases not matched: "b"
```

Also tried with the marker as a leading line comment above `default:`, as a leading block
comment, and as a block comment inside the clause body; all four report.

#### What the option actually does

It matches upstream typescript-eslint, where the comment is a *substitute for an omitted*
`default` clause rather than an annotation on a present one —
[`getCommentDefaultCase`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts)
reads the last comment *after the last case* and feeds it into `defaultCase ?? …`. So the
working shape has no `default` clause at all, and — because the suppression is
`considerDefaultExhaustiveForUnions && defaultCase != null` — needs the other option too:

```jsonc
{ "considerDefaultExhaustiveForUnions": true, "defaultCaseCommentPattern": "^skip default$" }
```

```ts
declare const value: 'a' | 'b';

switch (value) {
  case 'a':
    break;
  // skip default
}
```

Clean. Changing only the comment to something the pattern does not match brings the
finding back, so the comment is doing the work rather than riding on
`considerDefaultExhaustiveForUnions`. The implementation is fine — this is the behaviour
[#24338](https://github.com/oxc-project/oxc/issues/24338) asked for and
[tsgolint#1070](https://github.com/oxc-project/tsgolint/pull/1070) shipped.

#### Why it is worth fixing

The documented example is the exact case the option cannot serve, and the failure is
silent: the option is schema-valid (an unknown key is rejected, so a typo is not the
explanation), and the rule just keeps reporting. We spent a while concluding the option
was unimplemented before finding #24338 and re-probing.

The trap is worse than a wasted afternoon on a codebase where the `default` arm does
work. Taking the docs at their word suggests deleting `default: return ''` and leaving a
marker comment — which satisfies the rule while the same fallback moves to a `return ''`
after the switch. The hole survives and the guard now says it does not.

#### Suggested wording

```rust
/// Regular expression pattern for a comment that may take the place of an omitted
/// `default` clause. The comment must be the last one after the final case, and the
/// switch must have no `default` clause of its own; for union types this suppresses
/// the check only together with `considerDefaultExhaustiveForUnions`.
/// Example: `"^skip default$"` to allow a switch ending in `// skip default`.
```

Happy to send this as a PR if that is easier than an issue.
